### ISSUE TITLE
Enabling plugin causes WSoD in browsers

### ISSUE THREAD

**Issue by s1lviu** [mattermost/mattermost-plugin-zoom#33]
(now tracked in https://mattermost.atlassian.net/browse/MM-19287)

Plugin version:` Zoom (zoom - 1.0.7, pre-packaged)`

Mattermost info:
```
Version: 5.12.0
Build Number: 5.12.0
Build Date: Sat Jun 15 09:02:50 UTC 2019
Build Hash: bfd66aa445a2df8c6ed6ba9f2567021ecf6c9f3b
Build Enterprise Ready: true
DB Version: 5.12.0
```
With Zoom plugin enabled:

<img src="https://i.imgur.com/yjvh0sy.png">

When disabled, all the things are working ok.

The problem appears only on browsers. Tested on latest Chrome, Firefox with incognito.

In the mattermost desktop mac app, it works with plugin enabled.

I suppose that's a problem from the plugin caching system or mattermost's.

**Comment by cpoile** [comment]
Hi @s1lviu, can you watch the server log when this happens and cut and paste what you see there? Thanks!

**Comment by s1lviu** [comment]
Nothing strange. [Log](https://ybin.me/p/32548c4505f5aef9#0pWhNtzg+AghldGfotUCZJZG36x/8/POiW2Cs4bIMKQ=)

**Comment by cpoile** [comment]
@crspeller 
```
{"level":"error","ts":1562830169.1751978,"caller":"plugin/hclog_adapter.go:60","msg":"reading plugin stderr","plugin_id":"zoom","wrapped_extras":"errorread |0: file already closed"}
```

**Comment by crspeller** [comment]
> mkdir ./client/plugins: no such file or directory

I have seen this before when there is a permissions issue with the MM setup. Can you verify that the `client/plugins` folder exists and has the correct permissions?

**Comment by s1lviu** [comment]
```
root@mattermost:/opt/mattermost# ls -la client/plugins
total 24
drwxr--r-- 6 mattermost mattermost 4096 Jul 11 07:29 .
drwxr-xr-x 7 mattermost mattermost 4096 Jun 22 12:44 ..
drwxr--r-- 3 mattermost mattermost 4096 Jul  9 07:59 com.mattermost.draw-plugin
drwxr--r-- 2 mattermost mattermost 4096 Jun 22 12:42 com.mattermost.nps
drwxr--r-- 3 mattermost mattermost 4096 Feb 14 14:56 plugins~
drwxr--r-- 2 mattermost mattermost 4096 Jul 11 07:29 zoom
```

**Comment by dj5k** [comment]
We have this very same issue.
My current assumption is that this occurs when Mattermost is upgraded and the `./client/plugins` is not kept (which is what I did up until recently).
It appears that when a user was already logged in before the upgrade, upon opening the webclient it tries to restore the logged in state and requests the zoom plugin asset and if that does not exist for some reason mattermost returns 403 forbidden (I would have expected 404) and the page rendering stops.

This leaves me wondering whether Mattermost should handle this situation more gracefully. Or is this related to how the Zoom plugin manages its assets?

**Comment by levb** [comment]
Filed https://mattermost.atlassian.net/browse/MM-19287 to investigate

**Comment by limsammy** [comment]
Having this error as well, for all plugins. Also relevant forum post re related issue here: https://forum.mattermost.org/t/mattermost-logs-cli-command-not-working-ubuntu-18-04-3/8285

Here are my relevant logs:

```
{"level":"error","ts":1571114647.2104595,"caller":"plugin/hclog_adapter.go:60","msg":"reading plugin stderr","plugin_id":"github","wrapped_extras":"errorread |0: file already closed"}
{"level":"error","ts":1571114647.2129457,"caller":"plugin/hclog_adapter.go:60","msg":"reading plugin stderr","plugin_id":"com.mattermost.nps","wrapped_extras":"errorread |0: file already closed"}
```

Here's permissions when I am logged in with my main sudo user (not mattermost):
```
user@mattermost:/opt/mattermost$ ls -la client/plugins/
ls: cannot access 'client/plugins/.': Permission denied
ls: cannot access 'client/plugins/..': Permission denied
ls: cannot access 'client/plugins/com.mattermost.custom-attributes': Permission denied
ls: cannot access 'client/plugins/github': Permission denied
ls: cannot access 'client/plugins/com.github.manland.mattermost-plugin-gitlab': Permission denied
ls: cannot access 'client/plugins/jira': Permission denied
ls: cannot access 'client/plugins/com.mattermost.nps': Permission denied
ls: cannot access 'client/plugins/zoom': Permission denied
total 0
d????????? ? ? ? ?            ? .
d????????? ? ? ? ?            ? ..
d????????? ? ? ? ?            ? com.github.manland.mattermost-plugin-gitlab
d????????? ? ? ? ?            ? com.mattermost.custom-attributes
d????????? ? ? ? ?            ? com.mattermost.nps
d????????? ? ? ? ?            ? github
d????????? ? ? ? ?            ? jira
d????????? ? ? ? ?            ? zoom
```

And here's perms when running with mattermost user:

```
user@mattermost:/opt/mattermost$ sudo -u mattermost ls -la client/plugins/
total 0
drwxrw-r-- 1 mattermost mattermost  214 Oct 15 04:44 .
drwxrwxr-x 1 mattermost mattermost 4886 Oct 12 07:39 ..
drwxr--r-- 1 mattermost mattermost  140 Oct 15 04:43 com.github.manland.mattermost-plugin-gitlab
drwxr--r-- 1 mattermost mattermost  118 Oct 15 04:43 com.mattermost.custom-attributes
drwxr--r-- 1 mattermost mattermost   90 Oct 15 04:44 com.mattermost.nps
drwxr--r-- 1 mattermost mattermost   66 Oct 15 04:43 github
drwxr--r-- 1 mattermost mattermost   84 Oct 15 04:44 jira
drwxr--r-- 1 mattermost mattermost   62 Oct 15 04:44 zoom
```

**Comment by mickmister** [comment]
Hello all, it seems there are a few issues here:

1) The server failed to serve the bundle.

2) The webapp failed to gracefully handle the situation where the bundle was not loaded. This is fixed in 5.14 here: https://github.com/mattermost/mattermost-webapp/pull/3189 . The fix applies to the desktop app as well.

@limsammy do you know if this began occurring soon after a server upgrade or redeploy of the application? It seems that your experience happened recently so you may be able to provide the most context. Maybe a new plugin was installed around the time of its first occurrence?

I'm not sure what else would be trying to make changes to the directory that has the issue. The "file already closed" error is interesting. Does this happen when trying to fetch the plugin or when enabling it? I'm wondering if there is some more useful logs after changing the `LogSettings.FileLevel` to `DEBUG` temporarily. Note that this can slow down the server when there is a high volume of traffic.

> In the mattermost desktop mac app, it works with plugin enabled. I suppose that's a problem from the plugin caching system or mattermost's.

@s1lviu I think you are right. It makes sense the desktop app would have a cached version of the plugin bundle, and therefore loads the plugin with no problems.

**Comment by mickmister** [comment]
@limsammy Is this issue still occurring?

I'm going to close this issue for now. We can reopen if necessary.
